include/grpc/grpc_security_constants.h

```diff
@@ -74,26 +75,27 @@ typedef enum {
      presents a certificate.
 
      If the client presents a certificate, the client authentication is done by
-     grpc framework (The client needs to either present a signed cert or skip no
-     certificate for a successful connection).
+     the gRPC framework (the client needs to either present a signed cert or not
+     present a certificate at all for a successful connection).
 
      The key cert pair should still be valid for the SSL connection to be
      established. */
   GRPC_SSL_REQUEST_CLIENT_CERTIFICATE_AND_VERIFY,
-  /** Server requests client certificate but enforces that the client presents a
+  /** Server requests client certificate and enforces that the client presents a
      certificate.
 
      If the client presents a certificate, the client authentication is left to
-     the application based on the metadata like certificate etc.
+     the application (the necessary metadata will be available to the
+     application via authentication context properties, see grpc_auth_context).
 
      The key cert pair should still be valid for the SSL connection to be
      established. */
   GRPC_SSL_REQUEST_AND_REQUIRE_CLIENT_CERTIFICATE_BUT_DONT_VERIFY,
-  /** Server requests client certificate but enforces that the client presents a
+  /** Server requests client certificate and enforces that the client presents a
      certificate.
 
-     The cerificate presented by the client is verified by grpc framework (The
-     client needs to present signed certs for a successful connection).
+     The cerificate presented by the client is verified by the gRPC framework
+     (the client needs to present signed certs for a successful connection).
 
      The key cert pair should still be valid for the SSL connection to be
```

Comment:
I think the sentence "The key cert pair should still be valid for the SSL connection to be established" can be clearer.  Also, doesn't this statement apply to `GRPC_SSL_DONT_REQUEST_CLIENT_CERTIFICATE` as well?

Reply:
In GRPC_SSL_DONT_REQUEST_CLIENT_CERTIFICATE, server does not verify client certificate at all.

How about "The client's key and certificate themselves need to be valid for the SSL connection to be established". 

The main difference between VERIFY and BUT_DONT_VERIFY is whether the client certificate is chained back to the root certificates configured by the server. Except in GRPC_SSL_DONT_REQUEST_CLIENT_CERTIFICATE, the client key needs to match the client's certificate.